chainstate/tx-verifier/src/transaction_verifier/reward_distribution.rs

```diff
@@ -120,6 +136,51 @@ fn calculate_staker_reward(
     Some(staker_reward)
 }
 
+fn calculate_staker_reward_v1(
+    total_reward: Amount,
+    pool_balance: Amount,
+    staker_balance: Amount,
+    cost_per_block: Amount,
+    mpt: PerThousand,
+    pool_id: PoolId,
+) -> Result<Amount, ConnectTransactionError> {
+    debug_assert!(staker_balance <= pool_balance);
+
+    let staker_reward = match total_reward - cost_per_block {
+        Some(v) => {
+            let pool_balance = Uint256::from_amount(pool_balance);
+            let staker_balance = Uint256::from_amount(staker_balance);
+            let numer =
+                (Uint256::from_amount(v) * staker_balance).expect("Source types are smaller");
+            let pro_rata_staker_reward = (numer / pool_balance)
+                .ok_or(ConnectTransactionError::PoolBalanceIsZero(pool_id))?;
+            let pro_rata_staker_reward: AmountUIntType = pro_rata_staker_reward
+                .try_into()
+                .expect("cannot be greater than total_reward type");
+            let pro_rata_staker_reward = Amount::from_atoms(pro_rata_staker_reward);
+
+            let delegations_reward =
```

Comment:
delegators_reward, because this is owned by delegators.

Reply:
I think also from a type safety point of view, this should be Uint256, because then you multiply it by the margin ratio mpt, which will expand over u128. I know this is unlikely, but we should do it right anyway.